### PR TITLE
Check types in CI

### LINKED ISSUE #1072: type error in nav/nav-logo.tsx
@genehack could you look into this [type error](https://github.com/nextstrain/nextstrain.org/actions/runs/11826364406/job/32952189773#step:5:10)? It's related to changes from https://github.com/nextstrain/nextstrain.org/pull/1032

```
Error: components/nav/nav-logo.tsx(25,12): error TS2322: Type '{ src: string; }' is not assignable to type 'string | StaticImport'.
  Type '{ src: string; }' is missing the following properties from type 'StaticImageData': height, width
```

_Originally posted by @victorlin in https://github.com/nextstrain/nextstrain.org/pull/1070#discussion_r1841231752_

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

Previously, the only enforcement of TypeScript was through ESLint rules from @typescript-eslint.

## Related issues

- Closes #1072
- Closes #1080

## Checklist

- [x] Checks pass
- [x] Check if changes affect the [resource index JSON revision](https://docs.nextstrain.org/projects/nextstrain-dot-org/en/latest/resource-collection.html#resource-index-revisions)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "lint": "npm run lint:server",
     "lint:server": "DEBUG=eslint:cli-engine npx eslint --max-warnings=0 --ext .js,.jsx .",
     "lint:static-site": "cd static-site && DEBUG=eslint:cli-engine npx eslint --max-warnings=0 --ext .js,.jsx,.ts,.tsx .",
+    "type-check": "cd static-site && tsc",
```
Comment (victorlin) [inline package.json:20]: @genehack could you look into this [type error](https://github.com/nextstrain/nextstrain.org/actions/runs/11826364406/job/32952189773#step:5:10)? It's related to changes from https://github.com/nextstrain/nextstrain.org/pull/1032

```
Error: components/nav/nav-logo.tsx(25,12): error TS2322: Type '{ src: string; }' is not assignable to type 'string | StaticImport'.
  Type '{ src: string; }' is missing the following properties from type 'StaticImageData': height, width
```

Comment (genehack) [reply package.json:20]: yep, will try to look at that tomorrow.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -41,6 +41,17 @@ jobs:
...
+  type-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: 'package.json'
+      - run: npm ci
+      - run: npm run build
+      - run: npm run type-check
```
Comment (genehack) [inline .github/workflows/ci.yml:53]: if the type-checking depends on having a build, wouldn't it be better to include the build part in the `type-check` entry in `package.json:scripts`? 

IOW, make this only `npm run type-check` here and...

File: package.json (hunk 1 above)
Comment (genehack) [inline package.json:20]: ... `./build.sh && cd static-site && tsc` here?

**Review by genehack — APPROVED** [review]
non-blocking commentary.

**Comment by victorlin** [comment]
@genehack good point. I've added a separate `type-check:ci` script for this purpose. This lets us use `npm run type-check` in during local dev to skip the build – it takes ~1 minute to run and is not necessary all the time.

**Comment by genehack** [comment]
> @genehack good point. I've added a separate `type-check:ci` script for this purpose. This lets us use `npm run type-check` in during local dev to skip the build – it takes ~1 minute to run and is not necessary all the time.

👍 I was hoping that `./build.sh` was smart enough to only rebuild when needed. 8^/

**Pull request merged**